scripts/zones/Promyvion-Vahzl/mobs/Solicitor.lua

```diff
@@ -1,10 +1,34 @@
 -----------------------------------
 -- Area: Promyvion - Vahzl
 --   NM: Solicitor
+-- TODO: Verify cmbDelay
+-----------------------------------
+mixins = {require("scripts/mixins/families/empty_terroanima")}
 -----------------------------------
 local entity = {}
 
-entity.onMobDeath = function(mob, player, isKiller)
+entity.onMobSpawn = function(mob)
+    mob:setMod(xi.mod.DOUBLE_ATTACK, 20)
+end
+
+entity.onMobWeaponSkillPrepare = function(mob, target)
+    local roll = math.random()
+    if mob:getHPP() <= 35 then
+        if roll <= 0.7 then
+            return 1234 -- Carousel
+        else
+            return 1274 -- Impalement
+        end
+    end
+end
+
+entity.onMobFight = function(mob, target)
+    if mob:getTP() >= 2000 then
+        mob:useMobAbility()
+    end
+    if mob:getHPP() <= 35 then
+       mob:setMod(xi.mod.STORETP, 250)
```

Comment:
This is going to trigger repeatedly every tick, while you are using SET and not add so it isn't going to stack, it's still an uneeded execution. There has to be a better way to accomplish a low HP change in storeTP and if not, we could at least use a localvar

(this has me wondering if I can rig up latents for mobs)

Reply:
Not sure if you thought about this at all, I can stick on a localvar and call it a day. Sorry for the long awaited response and PRs. Had a little project I have been working on that needed my full attention :(

Reply:
I got busy with other things myself, so some of the older prs with changes requested didn't get a lot of checking back - I just try to check my github notifications to know if changes happened or not.